components/tidb_query_executors/src/runner.rs

```diff
@@ -510,6 +508,16 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
         self.out_most_executor.collect_storage_stats(dest);
     }
 
+    pub fn collect_scan_summary(&mut self, dest: &mut ExecSummary) {
+        if !self.collect_exec_summary {
+            return;
+        }
+        let result = &self.exec_stats.summary_per_executor;
+        if let Some(exec_stat) = result.first() {
```

Comment:
We should add some comments here, it's really confusing the first slot contains the summary of the scan executor if I didn't check the code of how the runner is constructed.